sql2parquet/sql2parquet.py

```diff
@@ -0,0 +1,120 @@
+        FileNotFoundError: If `config_file` doesn't exist
+        Exception: Any exception encountered except FileNotFoundError during reading TOML config.
+    '''
+
+    try:
+        return toml.load(open(config_file_name))
+    except FileNotFoundError:
+        logger.error('%s file doesn\'t exist!', config_file_name)
+        sys.exit('%s file doesn\'t exist!', config_file_name)
+    except Exception as e:
+        logger.error(e)
+        sys.exit(e)
+
+def sqlToParquet(config: Dict[str, str]) -> None:
+    '''
+    Read tables from remote SQL server and exports the data as parquet files to OUTPUT_DIRECTORY
+
+    Args:
+        config: Dictionary config object
+    '''
+
+    if not os.path.exists(OUTPUT_DIRECTORY): os.mkdir(OUTPUT_DIRECTORY)
+
+    for database in config['databases']:
+        for species in database['species']:
+
+            db_connection_str = f'mysql+pymysql://{database["db_user"]}@{database["location"]}:{database["port"]}/{species["db_name"]}'
+            engine = create_engine(db_connection_str)
+
+            for table in database['tables']:
+                logger.info('Exporting %s table for %s...', table["table_name"], species["species_name"])
+
+                df = pd.read_sql(table['query'], con=engine, params={'species_name': species['species_name']})
+
+                if not os.path.exists(os.path.join('parquet', f'data={table["table_name"]}')): os.mkdir(os.path.join('parquet', f'data={table["table_name"]}'))
+
+                full_directory_path = os.path.join('parquet', f'data={table["table_name"]}', f'species={species["species_name"]}')
+                if not os.path.exists(full_directory_path): os.mkdir(full_directory_path)
+
+                file_name = f'{species["db_name"]}-{table["table_name"]}{FILE_EXT}'
+                # default engine: pyarrow
+                df.to_parquet(os.path.join(full_directory_path, file_name))
+
+def uploadDirToS3() -> None:
+    '''
+    Uploads all parquet files in OUTPUT_DIRECTORY to AWS_S3_BUCKET
+
+    Raises:
+        ClientError: If boto3 faces any errors while uploading the parquet files
+    '''
+
+    s3_client = boto3.client('s3')
+
+    for dir in list(Path(OUTPUT_DIRECTORY).glob('**')):
+        files = glob.glob(os.path.join(dir, f'*{FILE_EXT}'))
+
+        # skip the iteration if no parquet files in the directory path
+        if not files: continue
+
+        for file in files:
+            # truncate current working directory path to get relative path
+            file = os.path.relpath(file, os.getcwd())
+            # put all the sub directories directly in the S3 bucket, not inside OUTPUT_DIRECTORY folder
+            awsPath = file.replace(os.path.join(OUTPUT_DIRECTORY,''), '')
+            # for Windows machines convert windows path to UNIX path
+            if os.name == 'nt': awsPath = str(Path(awsPath).as_posix())
+
+            try:
+                s3_client.upload_file(file, AWS_S3_BUCKET, awsPath)
+            except ClientError as e:
```

Comment:
The script dies silently when I didn't add the proper AWS credentials, logs show the following error
```
botocore.exceptions.NoCredentialsError: Unable to locate credentials
2022-07-04 14:01:16,053 DEBUG utils s3transfer.utils Releasing acquire 0/None
2022-07-04 14:01:16,053 ERROR sql2parquet __main__ Unable to locate credentials
```
I think it is worth capturing it.

Reply:
Handled NoCredentialsError exception in latest commit